lib/measured/measurable.rb

```diff
@@ -46,13 +48,17 @@ def inspect
   end
 
   def <=>(other)
-    if other.is_a?(self.class) && unit == other.unit
-      value <=> other.value
+    if other.is_a?(self.class)
+      other_converted = other.convert_to(unit)
+      value <=> other_converted.value
     end
   end
 
   def ==(other)
-    !!(other.is_a?(self.class) && unit == other.unit && value == other.value)
+    return false unless other.is_a?(self.class)
+
+    other_converted = other.convert_to(unit)
```

Comment:
If you are comparing on unit you don't need to convert other

`unit == other.unit && value == other.value`